pkg/context/notice.go

```diff
@@ -0,0 +1,63 @@
+package context
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	"github.com/Unknwon/com"
+	"github.com/gogs/gogs/pkg/setting"
+	"github.com/gogs/gogs/pkg/tool"
+	log "gopkg.in/clog.v1"
+)
+
+// readServerNotice checks if a notice file exists and loads the message to display
+// on all pages.
+func readServerNotice() map[string]interface{} {
+	fileloc := path.Join(setting.CustomPath, "notice")
```

Comment:
Should we consider add a subdirectory under custom path i.e. `notices`, and currently only use the file `notices/main.html`?

Reply:
I suppose that would be good.  Keeps the possibility open of adding more types of notices, or notices for specific pages.  Would using an html file be good?  Currently, the body text is rendered as html, but the header line is not, so it's not completely html content.

Just felt I should clarify that point, but I can change it to your suggestion if you want.

Reply:
>Currently, the body text is rendered as html, but the header line is not, so it's not completely html content.

You're right, actually I was thinking adding things like `title: My sweet title`, so we could possibly add more attributes in the future. And separate title and content by two line breaks (as right now, as how Git commit message works). What do you think?

Reply:
The commit message convention is exactly what I was thinking about too.  The code simply breaks off the first line and takes the rest as body text, but when I write my own notice files I always leave an empty line, so I agree it would be cleaner that way.

Another idea might be to render it as markdown (and force the file to be called `notice.md` to make it clear), so a short notice with a title could be:
```markdown
# Server downtime

We will be taking the server **offline** on Sunday.
```

Reply:
Markdown idea is actually brilliant! Would you mind update PR to address this?

Helpful link: https://github.com/gogs/gogs/blob/798636c95b207fc97a785ce545d7af7197361447/pkg/markup/markdown.go#L120